README.md

```diff
@@ -6,99 +6,155 @@
-      param :city, :string, required: true
-      param :street, :string
-      param :number, :integer
-    end
+  param do
+    param :city, :string, required: true
+    param :street, :string
+    param :number, :integer
   end
-  ```
-
-  Each field needs to define the type that will be parsed and the options (if any). The types available are:
-
-  - `:string`
-  - `:atom`
-  - `:integer`
-  - `:float`
-  - `:boolean`
-  - `{:map, inner_type}`
-  - `{:array, inner_type}`
-  - `:map`
-  - `:array`
-  - `:date`
-  - `:time`
-  - `:datetime`
-  - `:naive_datetime`
-  - `module`*
-
-  \\* Any module that implements the `Parameter.Field` behaviour is elegible to be a field in the schema definition.
-
-  The options available for the field definition are:
-  - `key`: This is the key on the external source that will be converted to the param definition.
-  As an example, if you receive data from an external source that uses snake case for mapping `first_name`, this flag should be `key: "firstName"`.
-  If this parameter is not set it will default to the field name.
-  - `default`: default value of the field.
-  - `required`: defines if the field needs to be present when parsing the input.
-
-  After the definition the schema can be validate and parsed against external parameters using the `Parameter.load/3` function.
-
-
-  ## Data Deserialization
-
-  This is a common requirement when you receive data from an external source and wants to
-  validate and deserialize this data to an Elixir definition. This can be achived using `Parameter.load/2` or `Parameter.load/3` functions:
-
-  ```elixir
-  iex> params = %{
-        "address" => %{"city" => "Tallinn"},
-        "age" => "32",
-        "firstName" => "Paulo",
-        "lastName" => "Curado"
-      }
-  ...> Parameter.load(UserSchema, params)
-    %{
-      address: %{city: "Tallinn"},
-      age: 32,
-      first_name: "Paulo",
-      last_name: "Curado"
+end
+```
+
+Each field needs to define the type that will be parsed and the options (if any). The types available are:
+
+- `:string`
+- `:atom`
+- `:integer`
+- `:float`
+- `:boolean`
+- `{:map, inner_type}`
+- `{:array, inner_type}`
+- `:map`
+- `:array`
+- `:date`
+- `:time`
+- `:datetime`
+- `:naive_datetime`
+- `module`*
+
+\\* Any module that implements the `Parameter.Field` behaviour is elegible to be a field in the schema definition.
+
+The options available for the field definition are:
+- `key`: This is the key on the external source that will be converted to the param definition.
+As an example, if you receive data from an external source that uses snake case for mapping `first_name`, this flag should be `key: "firstName"`.
+If this parameter is not set it will default to the field name.
+- `default`: default value of the field.
+- `required`: defines if the field needs to be present when parsing the input.
+
+After the definition the schema can be validate and parsed against external parameters using the `Parameter.load/3` function.
```

Comment:
After the definition, the schema can be validated and parsed against external parameters using the `Parameter.load/3` function.